### PR TITLE
rustscan: init at 1.8.0

### PR DESCRIPTION
**Pull request by SuperSandro2000**
###### Motivation for this change

Get rustscan into nixpkgs.

###### Things done

- [ ] Tested using sandboxing ([nix.useSandbox](https://nixos.org/nixos/manual/options.html#opt-nix.useSandbox) on NixOS, or option `sandbox` in [`nix.conf`](https://nixos.org/nix/manual/#sec-conf-file) on non-NixOS linux)
- Built on platform(s)
   - [ ] NixOS
   - [ ] macOS
   - [x] other Linux distributions
- [ ] Tested via one or more NixOS test(s) if existing and applicable for the change (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
- [ ] Tested compilation of all pkgs that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review wip"`
- [x] Tested execution of all binary files (usually in `./result/bin/`)
- [ ] Determined the impact on package closure size (by running `nix path-info -S` before and after)
- [x] Ensured that relevant documentation is up to date
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/.github/CONTRIBUTING.md).

### CODE REVIEW

File: maintainers/maintainer-list.nix [hunk 1]
```diff
@@ -7995,6 +7995,12 @@
     githubId = 65870;
     name = "Сухарик";
   };
+  SuperSandro2000 = {
+    email = "sandro.jaeckel@gmail.com";
+    github = "SuperSandro2000";
+    githubId = 7258858;
+    name = "Sandro Jäckel";
+  };
```
Comment (danieldk) [inline maintainers/maintainer-list.nix:8003]: Do this is in a separate commit with the message

```
maintainers: add SuperSandro2000
```

File: pkgs/tools/security/rustscan/default.nix [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+    sha256 = "0rkqsh4i58cf18ad97yr4f68s5jg6z0ybz4bw8607lz7cjkfvjay";
+  };
+
+  cargoSha256 = "0mj214f2md7kjknmcayc5dcfmlk2b8mqkn7kxzdis8qv9a5xcbk8";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  fixupPhase = ''
+    wrapProgram $out/bin/rustscan --prefix PATH ":" ${nmap}/bin ;
+  '';
```
Comment (danieldk) [inline pkgs/tools/security/rustscan/default.nix:25]: From an admittedly very quick look, it seems that `nmap` is only hardcoded in the source once. You could patch this up and then you do not need a wrapper (which is always nice):

```
  postPatch = ''
    substituteInPlace src/main.rs \
      --replace 'Command::new("nmap")' 'Command::new("$out/bin/nmap")'
  '';
```

Untested, but just to give an idea of how you could do this.

File: pkgs/tools/security/rustscan/default.nix [hunk 3]
```diff
@@ -0,0 +1,33 @@
+{ stdenv
```
Comment (danieldk) [inline pkgs/tools/security/rustscan/default.nix:1]:
```suggestion
{ lib
```

I saw that you are only using `stdenv` for meta. `lib` is more lightweight if only needed for that purpose.

File: pkgs/tools/security/rustscan/default.nix [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+
+  cargoSha256 = "0mj214f2md7kjknmcayc5dcfmlk2b8mqkn7kxzdis8qv9a5xcbk8";
+
+  nativeBuildInputs = [ makeWrapper ];
+
+  fixupPhase = ''
+    wrapProgram $out/bin/rustscan --prefix PATH ":" ${nmap}/bin ;
+  '';
+
+  meta = with stdenv.lib; {
```
Comment (danieldk) [inline pkgs/tools/security/rustscan/default.nix:27]:
```suggestion
  meta = with lib; {
```

(See above.)

File: pkgs/tools/security/rustscan/default.nix [hunk 5]
```diff
@@ -0,0 +1,33 @@
...
+    description = "Faster Nmap Scanning with Rust";
+    homepage = https://github.com/RustScan/RustScan;
```
Comment (danieldk) [inline pkgs/tools/security/rustscan/default.nix:29]: Unquoted URLs are deprecated.

```suggestion
    homepage = "https://github.com/RustScan/RustScan";
```

**Review by danieldk — CHANGES_REQUESTED** [review]
Thanks you for your first contribution :+1: .

I have added some requested changes.

File: pkgs/tools/security/rustscan/default.nix (hunk 2 above)
Comment (SuperSandro2000) [reply pkgs/tools/security/rustscan/default.nix]: Took me a few tries but I got it working. Thanks!

**Comment by danieldk** [comment]
Thanks for making the changes! I pushed a few more changes:

* Simplified `postPatch` a bit.
* `nmap` should not be in `nativeBuildInputs`.
* Added skips for some tests, because they fail in the nix sandbox.

**Comment by SuperSandro2000** [comment]
Thanks for the help! I am not running nixOS (yet?) so some of the tests are not that easy for me. I did a quick test and everything still works.

**Review by danieldk — APPROVED** [review]
Thanks, seems to work well :+1: .

**Comment by danieldk** [comment]
> Thanks for the help! I am not running nixOS (yet?) so some of the tests are not that easy for me.

If you enable Nix sandboxing, it should not make a different difference. Unless you are on macOS, where things are a bit trickier.

**Comment by SuperSandro2000** [comment]
> If you enable Nix sandboxing, it should not make a big difference. Unless you are on macOS, where things are a bit trickier.

I installed nix on Debian with the help of the package ``nix-setup-systemd``. It has sandboxing disabled but as far as I can tell it is only because it fails on aarch64. I am on x84_amd64 so I will probably make my next contributions with sandboxing enabled. Thanks for the heads up.
